core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1086,6 +1109,126 @@ class ParserTest extends munit.ScalaCheckSuite {
+      (genP, n, str) =>
+        val repA = genP.fa.repAs[NonEmptyVector[_]](n, n)
+        val repB = genP.fa.repNAs[NonEmptyVector[_]](n)
+        assertEquals(repA.parse(str), repB.parse(str), clue(repA, repB, str))
+    }
+  }
+
+  property("rep parses n entries, min <= n <= max") {
+    val validMinMax = for {
+      min <- Gen.choose(1, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+          assert(l.length >= min)
+        }
+      }
+    }
+  }
+
+  property("rep0 parses n entries, min <= n <= max") {
+    val validMinMax = for {
+      min <- Gen.choose(0, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep0(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+          assert(l.length >= min)
+        }
+      }
+    }
+  }
+
+  property("rep0 parses at most max entries (min == 0)") {
+    forAll(ParserGen.gen, Gen.choose(0, Int.MaxValue), Arbitrary.arbitrary[String]) {
+      (genP, max, str) =>
+        genP.fa.rep0(0, max).parse(str).foreach { case (_, l) => assert(l.length <= max) }
+    }
+  }
+
+  property("repAs parses max entries when available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * n
+      val p = Parser.char('a').repAs[Int](min = n, max = n)
+      val parsed = p.parseAll(toBeConsumed)
+      assertEquals(parsed, Right(n))
+    }
+  }
+
+  property("repAs parses max entries when more is available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * (n + 1)
+      val p = Parser.char('a').repAs[Int](min = 1, max = n)
+      val parsed = p.parse(toBeConsumed)
+      assertEquals(parsed, Right(("a", n)))
+    }
+  }
+
+  property("repAs0 parses max entries when available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * n
+      val p = Parser.anyChar.repAs0[Int](max = n)
+      val parsed = p.parseAll(toBeConsumed)
+      assertEquals(parsed, Right(n))
+    }
+  }
+
+  property("repAs0 parses max entries when more is available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * (n + 1)
+      val p = Parser.anyChar.repAs0[Int](max = n)
+      val parsed = p.parse(toBeConsumed)
+      assertEquals(parsed, Right(("a", n)))
+    }
+  }
+
+  property("rep parses max entries when available") {
+    forAll(Gen.choose(1, 100)) { (n: Int) =>
+      val toBeConsumed = "a" * n
+      val p = Parser.char('a').rep(min = n, max = n).map(_.length)
```

Comment:
can we use repN/exactly here?

Reply:
The difference between this one and the one above if that one is `rep` and the other is `repAs`. If we use exactly for both, then they become identical.